aspnetcore/fundamentals/http-context.md

```diff
@@ -0,0 +1,146 @@
+---
+title: Accessing HttpContext in ASP.NET Core
+author: coderandhiker
+description: Learn how to access HttpContext in ASP.NET Core.
+manager: wpickett
+ms.author: riande
+ms.custom: mvc
+ms.date: 05/22/2018
+ms.prod: aspnet-core
+ms.technology: aspnet
+ms.topic: article
+uid: fundamentals/httpcontext
+---
+# Use IHttpContextAccessor to manage HttpContext
+
+ASP.NET Core applications access the HttpContext through the [IHttpContextAccessor](/dotnet/api/microsoft.aspnetcore.http.ihttpcontextaccessor?view=aspnetcore-2.0) interface and its default implementation [HttpContextAccessor](/dotnet/api/microsoft.aspnetcore.http.httpcontextaccessor.httpcontext?view=aspnetcore-2.0).
+
+## Use the HttpContext from Razor Pages, controllers, and middleware
+
+The Razor Pages `PageModel` exposes the `HttpContext` property:
+
+```csharp
+public class AboutModel : PageModel
+{
+    public string Message { get; set; }
+
+    public void OnGet()
+    {
+        Message = HttpContext.Request.PathBase;
+    }
+}
+```
+
+Controllers expose the `HttpContext` property from the `ControllerBase`:
+
+```csharp
+public IActionResult About()
+{
+    var pathBase = HttpContext.Request.PathBase;
+    // Do something with the PathBase.
+
+    return View();
+}
+```
+
+When working with custom middleware components, `HttpContext` is passed into the `Invoke` method and can be accessed when the middleware is being configured.
+
+```csharp
+public class MyCustomMiddleware
+{
+    public async Task Invoke(HttpContext context)
+    {
+        // Middleware initialization optionally using HttpContext
+    }
+    ...
+}
+```
+
+## Use HttpContext from custom components
+
+For other framework and custom components that need access to `HttpContext`, the recommended approach is to register a dependency using the built-in [dependency injection](xref:fundamentals/dependency-injection) container. The dependency injection container will supply the `IHttpContextAccessor` to any classes that declare it as a dependency in their constructors.
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+     services.AddMvc();
+     services.AddSingleton<IHttpContextAccessor, HttpContextAccessor>();
```

Comment:
For 2.1+, this should ideally be using `AddHttpContextAccessor` which was added in https://github.com/aspnet/HttpAbstractions/pull/947/files